cmd/buildkitd/main.go

```diff
@@ -454,28 +452,30 @@ func applyMainFlags(c *cli.Context, cfg *config.Config, md *toml.MetaData) error
 	}
 
 	if c.IsSet("allow-insecure-entitlement") {
-		//override values from config
+		// override values from config
 		cfg.Entitlements = c.StringSlice("allow-insecure-entitlement")
 	}
 
 	if c.IsSet("debugaddr") {
 		cfg.GRPC.DebugAddress = c.String("debugaddr")
 	}
 
-	if md == nil || !md.IsDefined("grpc", "uid") {
```

Comment:
I think the intent here was to make these fields optional (being able to distinguish `0` (`root`) from "not set. I implemented that by changing the field in Config to a pointer, which (AFAICS) seems to work as expected, but let me know if there's concerns.